### DISCUSSION TITLE
Is it possible to use Depends from fastapi with this?

### DISCUSSION THREAD

**Discussion by chase-lewis-akasa** [miguelgrinberg/python-socketio#898] [Q&A]
I'm trying to access headers via FastAPI using this library. I can tell the headers ARE being sent by looking at the request. I just don't really see how i can hook this up to that system.

**Comment by miguelgrinberg** [comment]
This package integrates with web frameworks only at the WSGI/ASGI layer. There is a third-party integration package for FastAPI that you may want to research: https://github.com/pyropy/fastapi-socketio. I know nothing about it, though, please ask questions about this package to the author.

In any case, accessing the headers of the Socket.IO connection request is possible, and is explained in the [documentation](https://python-socketio.readthedocs.io/en/latest/server.html#connect-and-disconnect-event-handlers). The headers received in the Socket.IO connection are sent to you in the connection handler, in an `environ` argument. This is a dictionary that follows the WSGI specification format.
